mapie/calibration.py

```diff
@@ -0,0 +1,548 @@
+     [0.75432411 0.         0.        ]
+     [0.62285341 0.         0.        ]
+     [0.         0.66666667 0.        ]
+     [0.         0.66666667 0.        ]
+     [0.         0.66666667 0.        ]
+     [0.         0.         0.33333002]
+     [0.         0.         0.54326683]
+     [0.         0.         0.66666124]]
+    """
+
+    fit_attributes = [
+        "estimator",
+        "calibrators",
+    ]
+
+    valid_calibrators = {
+        "sigmoid": _SigmoidCalibration(),
+        "isotonic": IsotonicRegression(out_of_bounds="clip")
+    }
+
+    valid_methods = ["top_label"]
+
+    def __init__(
+        self,
+        estimator: Optional[ClassifierMixin] = None,
+        method: str = "top_label",
+        calibrator: Optional[Union[str, RegressorMixin]] = None,
+        cv: Optional[str] = "split",
+    ) -> None:
+        self.estimator = estimator
+        self.method = method
+        self.calibrator = calibrator
+        self.cv = cv
+
+    def _fit_calibrator(
+        self,
+        item: int,
+        calibrator: RegressorMixin,
+        y_calib: ArrayLike,
+        top_class_prob: NDArray,
+        y_pred: NDArray,
+        sample_weight: Optional[ArrayLike],
+    ) -> RegressorMixin:
+        """
+        Fitting the calibrator requires that for each class we
+        get the correct values to fit a calibrator for this specific
+        class.
+
+        Parameters
+        ----------
+        item : int
+            The class for which we will fit a calibrator.
+        calibrator : RegressorMixin
+            Calibrator to train.
+        y_calib : NDArray of shape (n_samples,)
+            Training labels.
+        top_class_prob : NDArray of shape (n_samples,)
+            The independent values of the calibrator, it represents the
+            maximum score in the probability predictions.
+        y_pred : NDArray of shape (n_samples,)
+            The array to determine which class the max prediction belongs to.
+        sample_weight : Optional[ArrayLike] of shape (n_samples,)
+            Sample weights. If None, then samples are equally weighted.
+
+        Returns
+        -------
+        RegressorMixin
+            Calibrated estimator.
+        """
+        calibrator_ = clone(calibrator)
+        y_calib = cast(NDArray, y_calib)
+        sample_weight = cast(NDArray, sample_weight)
+        correct_label = np.where(y_pred.ravel() == item)[0]
+        y_calib_ = check_binary_zero_one(
+            np.equal(y_calib[correct_label], item).astype(int)
+        )
+        top_class_prob_ = top_class_prob[correct_label]
+
+        if sample_weight is not None:
+            sample_weight_ = sample_weight[correct_label]
+            (
+                sample_weight_, top_class_prob_, y_calib_
+            ) = check_null_weight(  # type: ignore
+                sample_weight_,
+                top_class_prob_,
+                y_calib_
+            )
+        else:
+            sample_weight_ = sample_weight
```

Comment:
thus sample_weight_ is None ! What happens then?

Reply:
Then, this is delt with in `fit_estimator`.